control-plane/connect-inject/controllers/serviceaccount/serviceaccount_controller_test.go

```diff
@@ -135,15 +124,7 @@ func TestReconcile_CreateWorkloadIdentity(t *testing.T) {
 			expectedResource: &pbresource.Resource{
 				Id: &pbresource.ID{
 					Name: "my-svc-account",
-					Type: &pbresource.Type{
-						Group:        "auth",
-						GroupVersion: "v1alpha1",
-						Kind:         "WorkloadIdentity",
-					},
-					Tenancy: &pbresource.Tenancy{
-						Namespace: constants.DefaultConsulNS,
-						Partition: constants.DefaultConsulPartition,
-					},
```

Comment:
Not intentionally, but the test does pass for me locally 🤔

Reply:
I think Dan's right that it's from the smoothing changes R.B. and Semir have made - either way, feel free to cut out of this one, we can get them all in a follow-up.

Reply:
I think it passes because we only compare the data part of the workload identity instead of the whole resource. I tried to fix it but it fails with peer name from tenancy missing from tenancy so I think it needs to be addressed in a separate PR.

Reply:
Ah, thanks for the lead - and trying to fix. That makes sense 👍🏻